File: programmers/모의고사.js
function solution(answers) {
    let test1 = [1, 2, 3, 4, 5]
    let test2 = [2, 1, 2, 3, 2, 4, 2, 5]
    let test3 = [3, 3, 1, 1, 2, 2, 4, 4, 5, 5]
    let correct1 = 0
    let correct2 = 0
    let correct3 = 0
    let maxCorrect = 0
    let answer = []

    for (let i = 0; i < answers.length; i++) {
        if (test1[i % test1.length] === answers[i]) {
            correct1 += 1
            if (maxCorrect < correct1) {
                maxCorrect = correct1
            }
        }

        if (test2[i % test2.length] === answers[i]) {
            correct2 += 1
            if (maxCorrect < correct2) {
                maxCorrect = correct2
            }
        }

        if (test3[i % test3.length] === answers[i]) {
            correct3 += 1
            if (maxCorrect < correct3) {
                maxCorrect = correct3
            }
        }
    }

    if (maxCorrect === correct1) {
        answer.push(1)
    }
    if (maxCorrect === correct2) {
        answer.push(2)
    }
    if (maxCorrect === correct3) {
        answer.push(3)
    }

    return answer
}

// function solution(answers) {
//     let test = [[1, 2, 3, 4, 5], [2, 1, 2, 3, 2, 4, 2, 5], ]
//     let test2 = [2, 1, 2, 3, 2, 4, 2, 5]
//     let test3 = [3, 3, 1, 1, 2, 2, 4, 4, 5, 5]
//     let correct1 = 0
//     let correct2 = 0
//     let correct3 = 0
//     let maxCorrect = 0
//     let answer = []

//     for (let i = 0; i < answers.length; i++) {
//         if (test1[i % test1.length] === answers[i]) {
//             correct1 += 1
//             if (maxCorrect < correct1) {
//                 maxCorrect = correct1
//             }
//         }

//         if (test2[i % test2.length] === answers[i]) {
//             correct2 += 1
//             if (maxCorrect < correct2) {
//                 maxCorrect = correct2
//             }
//         }

//         if (test3[i % test3.length] === answers[i]) {
//             correct3 += 1
//             if (maxCorrect < correct3) {
//                 maxCorrect = correct3
//             }
//         }
//     }

//     if (maxCorrect === correct1) {
//         answer.push(1)
//     }
//     if (maxCorrect === correct2) {
//         answer.push(2)
//     }
//     if (maxCorrect === correct3) {
//         answer.push(3)
//     }

//     return answer
// }

File: programmers/모의고사.py
def solution(answers):
    test1 = [1, 2, 3, 4, 5]
    test2 = [2, 1, 2, 3, 2, 4, 2, 5]
    test3 = [3, 3, 1, 1, 2, 2, 4, 4, 5, 5]
    answer = []
    answer1 = []
    answer2 = []
    answer3 = []
    for i, val in enumerate(answers):
        if val == test1[i%len(test1)]:
            answer1.append(val)
        if val == test2[i%len(test2)]:
            answer2.append(val)
        if val == test3[i%len(test3)]:
            answer3.append(val)
    max_answer = max(len(answer1), len(answer2), len(answer3))
    answer = []
    if max_answer == len(answer1):
        answer.append(1)
    if max_answer == len(answer2):
        answer.append(2)
    if max_answer == len(answer3):
        answer.append(3)
    return answer